labseu2-workout-tracker/server
# Ft view workout history

##  What does this PR do
This PR will enable the user to get the history of his workouts.

##  Description of tasks to be completed
- add error response for unmatched routes
- add migration to update workout-session table
- write tests for workout history
- write model to query for a user's workout history
- write controller for workoutHistory endpoint
- add workoutHistory route to app router
- remove previous workout history implementation

##  How should this manually be tested
- clone this repository
- checkout `ft-view-workout-history` branch
- run `yarn install` or `npm install`
- run `yarn server` or `npm run server`
- run
```
  npx knex migrate:latest
  npx knex seed:run
```
signup by sending a post request to /auth/signup with the following credentials
```
{
  username: "testUser",
  email: "user@email.com",
  password: "password"
}
```
- use the token provided to send a post request to `/workouts/{id}/start` to start a workout session
- send a post request to `/workouts/{id}/end` to end the workout session
- send a get request to `/workouts/history` to see the history of workouts sessions created for the logged-in user 
##  What are the relevant Trello board stories
[As a user, I want to be able to view the history of my workouts.](https://trello.com/c/h1FNpxEH/66-as-a-user-i-want-to-be-able-to-view-the-history-of-my-workouts#)
